doc/developer-notes.md

```diff
@@ -839,6 +840,37 @@ namespace {
 #endif // BITCOIN_FOO_BAR_H
 ```
 
+Construct On First Use idiom
+------------------------------
+
+It is not possible to control the order in which non-local static objects in *different* translation units are initialized.
+This is the so-called "`static` initialization order problem". It could cause the use of an object:
+- before it is constructed (on program startup)
+- after it is destructed (at program exit)
+
+Both cases lead to undefined behavior.
+
+To prevent the `static` initialization order problem, use the Construct On First Use idiom:
+
+```C++
+BCLog::Logger& LogInstance()
+{
+    // See doc/developers-notes.md#construct-on-first-use-idiom
+    static BCLog::Logger* g_logger{new BCLog::Logger()};
+    return *g_logger;
+}
+```
+
+*Rationale*: This approache guaranties that:
```

Comment:
approach guarantees

Reply:
Thank you! [Updated](https://github.com/bitcoin/bitcoin/pull/18881#issuecomment-624007651).

Reply:
oops, somehow that comment was published before the review...GitHub :man_shrugging: